File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    participant = sorted(participant)
    completion = sorted(completion)
    answer = "."
    for i in range(len(completion)):
        if participant[i] != completion[i]:
            answer = participant[i]
            return answer
    return participant[len(participant)-1]

participant =["marina", "josipa", "nikola", "vinko", "filipa"]	
completion = ["josipa", "filipa", "marina", "nikola"]	
answer = solution(participant, completion)
print('"{}"'.format(answer))

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    static Map<String, Integer> participants = new HashMap<>();
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        init(participant);
        endGame(completion);
        
        for(String name: participant){
            if(participants.get(name) > 0){
                answer = name;
                break;
            }
        }
        return answer;
    }
    public void endGame(String[] completion){
        for(String name: completion){
            int value = participants.get(name);
            participants.put(name, value-1);
        }
    }
    
    
    public void init(String[] participant){
        for(String name: participant){
            if(participants.containsKey(name)){
                int n = participants.get(name);
                participants.put(name, n+1);
                continue;
            }
            participants.put(name, 1);
        }
    }
}